app/src/main/java/com/fcaputo/parkingapp/mvp/view/SettingsView.kt

```diff
@@ -0,0 +1,70 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.R
+import com.fcaputo.parkingapp.databinding.ActivitySettingsBinding
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class SettingsView(activity: Activity) : ActivityView(activity),SettingsContract.View {
+    private var binding: ActivitySettingsBinding = ActivitySettingsBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.saveButton.setOnClickListener{
+            if(fieldsAreComplete()) {
+                onClick()
+            } else {
+                showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            }
+        }
+    }
+
+    private fun fieldsAreComplete(): Boolean = listOf(binding.textInputParkingSize).all{ it.editText?.text?.isNotEmpty() ?: false }
```

Comment:
this should be in a presenter, not here, otherwise you will never ve able to test it

Reply:
Done